backend/module/authn/authn.go

```diff
@@ -52,9 +55,58 @@ func (m *mod) Register(r module.Registrar) error {
 type api struct {
 	provider authn.Provider
 	issuer   authn.Issuer
+	logger   *zap.Logger
+}
+
+func (a *api) loginViaRefresh(ctx context.Context, redirectURL string) (*authnv1.LoginResponse, error) {
+	md, ok := metadata.FromIncomingContext(ctx)
+	if !ok {
+		return nil, nil
+	}
+	cookies := md.Get("grpcgateway-cookie")
+	if len(cookies) == 0 {
+		return nil, nil
+	}
+
+	refreshToken, err := mux.GetCookieValue(cookies, "refreshToken")
+	if err != nil {
+		return nil, nil
+	}
+
+	token, err := a.issuer.RefreshToken(ctx, &oauth2.Token{RefreshToken: refreshToken})
+	if err != nil {
+		return nil, err
+	}
+
+	err = grpc.SendHeader(ctx, metadata.New(map[string]string{
```

Comment:
should this be `SetHeader` instead of `SendHeader`?

Reply:
good catch, makes more sense to stick to Set instead of Send even though we're not setting headers elsewhere yet. will change other instances of it as well